pkg/provider/aws/infra_template_test.go

```diff
@@ -126,3 +127,197 @@ func TestInfraTemplateModification(t *testing.T) {
 		})
 	}
 }
+
+func Test_GenerateCloudfrontDistributions(t *testing.T) {
+	cases := []struct {
+		name    string
+		results []core.CloudResource
+		cfg     config.Application
+		data    TemplateData
+		want    []*resources.CloudfrontDistribution
+	}{
+		{
+			name: "simple gateway test",
+			results: []core.CloudResource{
+				&core.Gateway{
+					Name:   "gw",
+					GWType: core.GatewayKind,
+					Routes: []core.Route{{Path: "/"}},
+				},
+			},
+			cfg: config.Application{
+				Provider: "aws",
+				AppName:  "app",
+				Exposed: map[string]*config.Expose{
+					"gw": {
+						Type:                   "apigateway",
+						ContentDeliveryNetwork: config.ContentDeliveryNetwork{Id: "distro"},
+					},
+				},
+			},
+			data: TemplateData{
+				TemplateData: provider.TemplateData{
+					Gateways: []provider.Gateway{
+						{Name: "gw"},
+					},
+				},
+			},
+			want: []*resources.CloudfrontDistribution{
+				{
+					Id: "app-distro",
+					Origins: []core.ResourceKey{
+						{Kind: core.GatewayKind, Name: "gw"},
+					},
+				},
+			},
+		},
+		{
+			name: "simple static unit test",
+			results: []core.CloudResource{
+				&core.StaticUnit{
+					Name: "su",
+				},
+			},
+			cfg: config.Application{
+				Provider: "aws",
+				AppName:  "app",
+				StaticUnit: map[string]*config.StaticUnit{
+					"su": {
+						Type:                   "apigateway",
+						ContentDeliveryNetwork: config.ContentDeliveryNetwork{Id: "distro"},
+					},
+				},
+			},
+			data: TemplateData{
+				TemplateData: provider.TemplateData{
+					StaticUnits: []provider.StaticUnit{
+						{Name: "su"},
+					},
+				},
+			},
+			want: []*resources.CloudfrontDistribution{
+				{
+					Id: "app-distro",
+					Origins: []core.ResourceKey{
+						{Kind: core.StaticUnitKind, Name: "su"},
+					},
+				},
+			},
+		},
+		{
+			name: "simple static unit with index document test",
+			results: []core.CloudResource{
+				&core.StaticUnit{
+					Name:          "su",
+					IndexDocument: "index.html",
+				},
+			},
+			cfg: config.Application{
+				Provider: "aws",
+				AppName:  "app",
+				StaticUnit: map[string]*config.StaticUnit{
+					"su": {
+						Type:                   "apigateway",
+						ContentDeliveryNetwork: config.ContentDeliveryNetwork{Id: "distro"},
+					},
+				},
+			},
+			data: TemplateData{
+				TemplateData: provider.TemplateData{
+					StaticUnits: []provider.StaticUnit{
+						{Name: "su"},
+					},
+				},
+			},
+			want: []*resources.CloudfrontDistribution{
+				{
+					Id: "app-distro",
+					Origins: []core.ResourceKey{
+						{Kind: core.StaticUnitKind, Name: "su"},
+					},
+					DefaultRootObject: "index.html",
+				},
+			},
+		},
+		{
+			name: "static unit and gw test",
```

Comment:
yeah see comment above, lets have a chat about this because this is purely how we want user experience to be and thats the main part i wanted some feedback on